particles/Tutorial/Javascript/Demo/source/MoveApplier.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/* global defineParticle */
+
+defineParticle(({SimpleParticle, log}) => {
+
+  const winningSequences = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6]
+  ];
+
+  return class extends SimpleParticle {
+    update({move, gameState}) {
+      if (!gameState || (move && move.type == 'reset' && gameState.lastMove != -1)) {
+        const gs = {
+          board: JSON.stringify(['', '', '', '', '', '', '', '', '']),
+          gameOver: false,
+          winnerAvatar: null,
+          currentPlayer: Math.floor(Math.random() * 2),
+          lastMove: -1,
+        };
+        this.set('gameState', gs);
+        return;
```

Comment:
As mentioned, I like to avoid control-flow statements (aka `return`) in `update/render`. Generally if the state conditionals are correct, you won't get improper behavior, performance can be improved, and it's easier to read the logic.